runelite-client/src/main/java/net/runelite/client/plugins/hiscore/HiscorePanel.java

```diff
@@ -471,13 +479,29 @@ private void applyHiscoreResult(HiscoreResult result)
 			}
 			else if ((s = result.getSkill(skill)) != null)
 			{
+				label.setForeground(Color.WHITE);
 				final long exp = s.getExperience();
 				final boolean isSkill = skill.getType() == HiscoreSkillType.SKILL;
-				int level = -1;
-				if (config.virtualLevels() && isSkill && exp > -1L)
+				long level = -1;
+				if (config.maxedSkillsStyle() == MaxedSkillsStyle.VIRTUAL_LEVELS && isSkill && exp > -1L)
 				{
 					level = Experience.getLevelForXp((int) exp);
 				}
+				else if (config.maxedSkillsStyle() == MaxedSkillsStyle.SHOW_EXPERIENCE && (isSkill || skill.getType() == HiscoreSkillType.OVERALL) && exp > -1L)
+				{
+					if ((isSkill && s.getLevel() == 99) || s.getLevel() == 2277)
+					{
+						level = exp;
+						label.setForeground((isSkill && exp == 200_000_000) || exp == 4_600_000_000L
```

Comment:
`4_600_000_000L` maxed xp will change when sailing comes out

Reply:
I updated it to use net.runelite.api.Skill.values().length. This enum is used all over the place and will surely be updated.